server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/disk/schemafile/SchemaFile.java

```diff
@@ -894,30 +911,30 @@ static short reEstimateSegSize(int oldSize) {
    */
   static short estimateSegmentSize(IMNode node) {
     int childNum = node.getChildren().size();
-    int tier = SchemaFile.SEG_SIZE_LST.length;
+    int tier = SchemaPage.SEG_SIZE_LST.length;
     if (childNum > 300) {
-      return SchemaFile.SEG_SIZE_LST[tier - 1] > SchemaFile.SEG_MIN_SIZ
-          ? SchemaFile.SEG_SIZE_LST[tier - 1]
-          : SchemaFile.SEG_MIN_SIZ;
+      return SchemaPage.SEG_SIZE_LST[tier - 1] > SchemaPage.SEG_MIN_SIZ
+          ? SchemaPage.SEG_SIZE_LST[tier - 1]
+          : SchemaPage.SEG_MIN_SIZ;
     } else if (childNum > 150) {
-      return SchemaFile.SEG_SIZE_LST[tier - 2] > SchemaFile.SEG_MIN_SIZ
-          ? SchemaFile.SEG_SIZE_LST[tier - 2]
-          : SchemaFile.SEG_MIN_SIZ;
+      return SchemaPage.SEG_SIZE_LST[tier - 2] > SchemaPage.SEG_MIN_SIZ
+          ? SchemaPage.SEG_SIZE_LST[tier - 2]
+          : SchemaPage.SEG_MIN_SIZ;
     } else if (childNum > 75) {
-      return SchemaFile.SEG_SIZE_LST[tier - 3] > SchemaFile.SEG_MIN_SIZ
-          ? SchemaFile.SEG_SIZE_LST[tier - 3]
-          : SchemaFile.SEG_MIN_SIZ;
+      return SchemaPage.SEG_SIZE_LST[tier - 3] > SchemaPage.SEG_MIN_SIZ
+          ? SchemaPage.SEG_SIZE_LST[tier - 3]
+          : SchemaPage.SEG_MIN_SIZ;
     } else if (childNum > 40) {
-      return SchemaFile.SEG_SIZE_LST[tier - 4] > SchemaFile.SEG_MIN_SIZ
-          ? SchemaFile.SEG_SIZE_LST[tier - 4]
-          : SchemaFile.SEG_MIN_SIZ;
+      return SchemaPage.SEG_SIZE_LST[tier - 4] > SchemaPage.SEG_MIN_SIZ
+          ? SchemaPage.SEG_SIZE_LST[tier - 4]
+          : SchemaPage.SEG_MIN_SIZ;
     } else if (childNum > 20) {
-      return SchemaFile.SEG_SIZE_LST[tier - 5] > SchemaFile.SEG_MIN_SIZ
-          ? SchemaFile.SEG_SIZE_LST[tier - 5]
-          : SchemaFile.SEG_MIN_SIZ;
+      return SchemaPage.SEG_SIZE_LST[tier - 5] > SchemaPage.SEG_MIN_SIZ
+          ? SchemaPage.SEG_SIZE_LST[tier - 5]
+          : SchemaPage.SEG_MIN_SIZ;
     }
```

Comment:
Show we implement this as a table driven process? The huge ```if-else``` is hard to maintain.

Reply:
It has been replaced with a table driven like process, hope it makes sense.